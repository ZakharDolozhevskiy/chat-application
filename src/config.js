export const api = {
  host: 'http://185.13.90.140:8081/',
  channel: 'message'
};

export const routes = {
  MAIN: '/',
  SETTINGS: '/settings'
};