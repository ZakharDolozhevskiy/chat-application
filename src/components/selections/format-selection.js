import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Radio from '@material-ui/core/es/Radio/Radio';
import Typography from '@material-ui/core/Typography';
import RadioGroup from '@material-ui/core/es/RadioGroup/RadioGroup';
import FormControlLabel from '@material-ui/core/es/FormControlLabel/FormControlLabel';

import { baseStyles } from './styles';
import { timeFormats } from '../../config';

class FormatSelection extends React.PureComponent {
  onChange = (event, theme) =>
    this.props.onChange(theme);

  render() {
    const { className, translate, timeFormat } = this.props;

    return (
      <div className={className}>
        <Typography component="p" color="inherit" className="label">
          {translate('Clock displays')}:
        </Typography>
        <RadioGroup value={timeFormat} onChange={this.onChange} className="controls">
          <FormControlLabel
            value={timeFormats[12]}
            label={translate('12 Hours')}
            control={<Radio className="selection-radio"/>}
          />
          <FormControlLabel
            value={timeFormats[24]}
            label={translate('24 Hours')}
            control={<Radio className="selection-radio"/>}
          />
        </RadioGroup>
      </div>
    );
  }
}

FormatSelection.propTypes = {
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string,
  translate: PropTypes.func.isRequired,
  timeFormat: PropTypes.string.isRequired
};

export default styled(FormatSelection)`${baseStyles}`;
