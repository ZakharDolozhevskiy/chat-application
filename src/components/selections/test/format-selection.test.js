import React from 'react';
import toJson from 'enzyme-to-json';
import { FormatSelection } from '../format-selection';
import { shallow } from 'enzyme';

import { timeFormats } from '../../../config';

describe('<FormatSelection/>', () => {
  const callback = jest.fn();

  const component = shallow(
    <FormatSelection
      className=""
      onChange={callback}
      translate={id => id}
      timeFormat={timeFormats[24]}
    />
  );

  it('should render component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call onChange callback if changes happen', () => {
    component.instance().onChange(null, timeFormats[12]);
    expect(callback.mock.calls[0][0]).toBe(timeFormats[12]);
  });
});