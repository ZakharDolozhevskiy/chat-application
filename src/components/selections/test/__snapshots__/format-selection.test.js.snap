// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatSelection/> should render component 1`] = `
<div
  className=""
>
  <WithStyles(Typography)
    className="label"
    color="inherit"
    component="p"
  >
    Clock displays
    :
  </WithStyles(Typography)>
  <RadioGroup
    className="controls"
    onChange={[Function]}
    value="HH:mm"
  >
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Radio)
          className="selection-radio"
        />
      }
      label="12 Hours"
      value="hh:mm a"
    />
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Radio)
          className="selection-radio"
        />
      }
      label="24 Hours"
      value="HH:mm"
    />
  </RadioGroup>
</div>
`;
